src/ngraph/runtime/cpu/pass/cpu_workspace_insertion.cpp

```diff
@@ -52,7 +52,9 @@
 #include "ngraph/runtime/cpu/op/max_pool_with_indices.hpp"
 #include "ngraph/runtime/cpu/op/sigmoid.hpp"
 
-void ngraph::runtime::cpu::pass::CPUWorkspaceInsertion::construct_max_pool_with_indices()
+using namespace ngraph;
+
+static std::shared_ptr<pattern::Matcher> create_maxpool_with_indices_matcher()
```

Comment:
Do we need pattern matcher here? Why not just look for MaxPoolBackprop ops?

Reply:
It's being used here as a parameter object pattern i.e. it groups together multiple arguments to pass into `transform`.